x-pack/plugins/infra/server/lib/domains/log_entries_domain/log_entries_search_strategy.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TSearchStrategyProvider } from 'src/plugins/data/server';
+import {
+  logEntriesRequestRT,
+  LOG_ENTRIES_SEARCH_STRATEGY,
+  logEntriesResponseRT,
+} from '../../../../common/log_entries';
+import { decodeOrThrow } from '../../../../common/runtime_types';
+
+export const logEntriesSearchStrategyProvider: TSearchStrategyProvider<typeof LOG_ENTRIES_SEARCH_STRATEGY> = ({
+  data,
+  logSources,
+}) => ({
+  search: async (request) => {
+    const payload = decodeOrThrow(logEntriesRequestRT)(request);
+
+    // something using data.search.search and payload
+    const { firstTotal, firstLoaded, firstResults } = await somethingThatCallsAnotherSearchStrategy(
+      payload
+    );
+
+    // something using data.search.search and the first results
+    const {
+      secondTotal,
+      secondLoaded,
+      secondResults,
+    } = await somethingThatCallsAnotherSearchStrategy(firstResults);
+
+    const total = mergeTotalsSomehow(firstTotal, secondTotal);
+    const loaded = mergeLoadedSomehow(firstLoaded, secondLoaded);
```

Comment:
This is meant to symbolize a sequential composition of two calls to other search strategies. Ideally parallel composition would also be possible.

Reply:
Supporting streaming results in the general case will also be tricky.

Reply:
After some more thought providing combinators for the general case us probably not feasible. Maybe thoroughly documenting the runtime semantics and response shape would be sufficient to enable devs to create specific composed strategies.